src/vs/base/common/glob.ts

```diff
@@ -526,6 +526,50 @@ export function getPathTerms(patternOrExpression: ParsedPattern | ParsedExpressi
 }
 
 function parsedExpression(expression: IExpression, options: IGlobOptions): ParsedExpression {
+	const exclude: IExpression = {};
+	const include: IExpression = {};
+
+	for (let key of Object.getOwnPropertyNames(expression)) {
+		if (expression[key] === false) {
+			include[key] = true;
+		} else {
+			exclude[key] = expression[key];
+		}
+	}
+
+	const excludeExpr = parsedSameValueExpression(exclude, options);
+	if (Object.getOwnPropertyNames(include).length === 0) {
+		return excludeExpr;
+	}
+	const includeExpr = parsedSameValueExpression(include, options);
```

Comment:
Can we add some new lines and a bit of comments to make this easier to read and understand.

Reply:
Will add comments. Thanks for the review.